docs/contrib/DeveloperGuide.md

```diff
@@ -134,6 +134,13 @@ examples. Use swig to define internal functions to C++ functionality.
 Tcl files can be included by encoding them in CMake into a string that
 is evaluated at run time (See [`Resizer::init()`](../main/src/rsz/src/Resizer.cc)).
 
+:::{Note}
+Please refer to the top-level Tcl formatting [guide](TclFormat.md).
+Our top-level Tcl files, in particular, have to be formatted in this specific
+manner because of the automatic parsing used to convert the READMEs into
+manpages.
```

Comment:
Is there a way to add a syntax check to the CI? That way it will fail if any changes don't adhere to the required style?

Reply:
We can do that, currently we have 2 syntax checkers in the regression 
1. `man_tcl_check.py` - check for function name existing in 
2. `man_tcl_params.py` - check for function params existing in help, proc and readme. Tests can be waived if `;# checker off` is added to the relevant `sta::define_cmd_args`/ `sta::parse_key_args` lines (example: check_antennas where there are deprecated args that appear in proc but not help)

I am guessing we can move the second one to the CI and remove it from regression test?

Reply:
As long as there is a check to make sure documentation stays updated it sounds good to me 👍

Reply:
CI results: https://github.com/luarss/OpenROAD/actions/runs/7787611080/job/21235152193